lib/Mir.ml

```diff
@@ -65,49 +63,39 @@ and ('e, 's) statement =
   (* An SList does not share any of Block's semantics - it is just multiple
      (ordered!) statements*)
   | SList of 's list
-  | Decl of {decl_adtype: autodifftype; decl_id: string; decl_type: unsizedtype}
+  | Decl of
+      { decl_adtype: autodifftype
+      ; decl_id: string
+      ; decl_type: 'e sizedtype }
```

Comment:
Could you also add an unsized decl here?

Reply:
Can I let you do that since you need it for the optimizer?

Reply:
Sure. There would need to be code gen for it at some point though. I guess we can figure that out later.